### DISCUSSION TITLE
Slow performance on repeating tasks

### DISCUSSION THREAD

**Discussion by boxallar** [TechbeeAT/jtxBoard#1294] [General]
I might be asking the app to do something it was not intended for.
 
I have a few notes and journals and lots of tasks. This is why I needed a good app. 
Recently, I wanted to add some daily recurring tasks. I selected "ends never" but it was so slow, I gave up and tried "ends after 60 days". 
I have 7 tasks like this and the performance has slowed a lot. Closing other apps or restarting my phone does not help. 
The app opens snappily still, but switching between the journals, notes and tasks tabs takes at least a minute for the entries to load. Similarly, editing or creating entries takes a long time as does saving and syncing. 
I will find another way to deal with my recurring todos, but such an extreme drop in performance might need to be looked at before peoples jtx-boards fill up.

**Comment by patrickunterwegs** [comment]
Hi @boxallar !
I agree with you, with 2000+ entries (which happens easily when you have a few recurring entries), the app gets a bit painful to use. I have invested a lot of time to find the actual cause and made some improvements in 2.7.2, but it was not enough. I just released the version now to get a bit forward (and I didn't want to block other functional extensions from being released), but I still have some refactoring to do. 
Could you give some feedback if you saw a bit of performance improvement from 2.7.1 to 2.7.2?

**Reply by boxallar** [reply]
Thanks for the quick response. I appreciate your hard work. I did not notice any difference at the time. Looking back, there might have been a bit of an improvement. It might be my mind playing tricks on me though.

**Reply by patrickunterwegs** [reply]
Thanks for the feedback!
I will focus on more refactoring in the upcoming months to increase efficiency. 
I have added a lot of features and I tried to stick to a certain software architecture, it seems like this was not always a good decision. I think I need to redo a few parts.
